### PR TITLE
Use context managers for disable_errors()

### PR DESCRIPTION
**Pull request by dixith**
### Description
This PR fixes one of the issue reported in #1184

## Test Plan
Since this change is coding style change, new cases are not added.

### CODE REVIEW

File: mypy/messages.py [hunk 1]
```diff
@@ -136,8 +137,11 @@ def add_errors(self, messages: 'MessageBuilder') -> None:
                 for info in errs:
                     self.errors.add_error_info(info)
 
-    def disable_errors(self) -> None:
+    @contextmanager
+    def disable_errors(self) -> Iterator[None]:
         self.disable_count += 1
+        yield
```
Comment (JelleZijlstra) [inline mypy/messages.py:143]: Should use try-finally

File: mypy/messages.py [hunk 2]
```diff
@@ -136,8 +137,11 @@ def add_errors(self, messages: 'MessageBuilder') -> None:
...
+        self.disable_count -= 1
 
     def enable_errors(self) -> None:
```
Comment (JelleZijlstra) [inline mypy/messages.py:146]: Can this method be removed?

**Review by JelleZijlstra — CHANGES_REQUESTED** [review]
Thanks for your contribution!

File: mypy/messages.py (hunk 2 above)
Comment (dixith) [reply mypy/messages.py]: Removed enable_errors()

File: mypy/messages.py (hunk 1 above)
Comment (dixith) [reply mypy/messages.py:143]: Thank you for the review @JelleZijlstra 

yield should be inside try-finally? 

`@contextmanager
    def disable_errors(self) -> Iterator[None]:
        self.disable_count += 1
        try:
            yield
        finally:
            pass
        self.disable_count -= 1`

Comment (JelleZijlstra) [reply mypy/messages.py:143]: And the `-= 1` should be in the finally block. The point is to make sure the cleanup code always happens.

Comment (dixith) [reply mypy/messages.py:143]: Got it.

Comment (dixith) [reply mypy/messages.py]: @JelleZijlstra Now that we have try-finally inside `enable_errors()` below try-finally is redundent right? Can be removed? https://github.com/python/mypy/blob/8860575a2134d87d890119015a9ead172f7fdfb1/mypy/checkexpr.py#L2811

```python
if right_map is None:
  with self.msg.disable_errors():
     try:
        right_type = self.analyze_cond_branch(right_map, e.right, left_type)
     finally:
        pass
```

**Comment by dixith** [comment]
Getting below error. I should use `TypeMap` type for `right_map`?
```mypy/checkexpr.py:2772: error: Local variable 'right_map' has inferred type None; add an annotation```

File: mypy/messages.py (hunk 1 above)
Comment (JelleZijlstra) [reply mypy/messages.py]: A try-finally with an empty finally block doesn't do anything useful, so yes you can remove it.

**Comment by dixith** [comment]
@JelleZijlstra 

Getting below error. Do you ahve any suggestions? 
```mypy/checkexpr.py:2772: error: Local variable 'right_map' has inferred type None; add an annotation```

**Comment by JelleZijlstra** [comment]
Sorry, I'm not too familiar with that code. I'd try adding an annotation though, as the error message suggests.

**Comment by JelleZijlstra** [comment]
Sorry for dropping this! We ended up merging #10569 instead.

**Pull request closed without merging**